src/Sql/SqlPgsql.php

```diff
@@ -43,7 +43,9 @@ private function createPasswordFile()
 
     public function command()
     {
-        return 'psql -q';
+        // Stop as soon as there's an error anywhere in the script. Ensures the
+        // appropriate exit code is used upon failure.
+        return 'psql -v ON_ERROR_STOP=1 -q';
```

Comment:
Yes, this is an unrelated bug that was highlighted during the test. The [default exit status](https://www.postgresql.org/docs/10/app-psql.html#id-1.9.4.18.7) of `psql` is always 0 whenever it executes a script even if it fails e.g. when the `{}` identifier wrappers are present.

https://serverfault.com/q/22833/394973

> By default, command processing continues after an error. When this variable is set to on, processing will instead stop immediately.

However, this current implementation is a breaking change as it stops imports of dumps ran without the `--if-exists` as they would be missing their appropriate `IF EXISTS` condition.
So in an iteration I had a couple of weeks back before I got busy with other stuff, I had switched to setting it as a prefix in the `queryFormat` method (which only gets called when you're passing in a custom SQL query, maintaining backwards-compatibility).

```patch
diff --git a/src/Sql/SqlPgsql.php b/src/Sql/SqlPgsql.php
--- a/src/Sql/SqlPgsql.php
+++ b/src/Sql/SqlPgsql.php
@@ -43,9 +43,7 @@
 
     public function command()
     {
-        // Stop as soon as there's an error anywhere in the script. Ensures the
-        // appropriate exit code is used upon failure.
-        return 'psql -v ON_ERROR_STOP=1 -q';
+        return 'psql -q';
     }
 
     public function getEnv()
@@ -110,7 +108,17 @@
         if (strtolower($query) == 'show tables;') {
             return PSQL_SHOW_TABLES;
         }
-        return $query;
+        /**
+         * @todo psql _should_ ideally return an error exit code by default
+         *   on the script level via the "-v ON_ERROR_STOP=ON" option:
+         *   https://serverfault.com/q/22833/394973
+         *   But can't be addressed at the moment in order to maintain backwards
+         *   compatibility, as it might no longer be able to import dumps
+         *   without the "IF EXISTS" conditions:
+         */
+        // Stop as soon as there's an error anywhere in the custom query.
+        // Ensures that the appropriate exit code is used upon failure.
+        return "\set ON_ERROR_STOP true\n" . $query;
     }
 
     public function listTables()
@@ -133,12 +141,8 @@
 
         $create_db = $this->getOption('create-db');
 
-        $environment = "";
-        $pw_file = $this->createPasswordFile();
-        if (isset($pw_file)) {
-            $environment = "PGPASSFILE={$pw_file} ";
-        }
-        $exec = "{$environment}pg_dump ";
+        $pg_dump_command = "pg_dump --if-exists";
+        $exec = $pg_dump_command;
         // Unlike psql, pg_dump does not take a '--dbname=' before the database name.
         $extra = str_replace('--dbname=', ' ', $this->creds());
         if ($data_only) {
@@ -166,7 +170,7 @@
                 foreach ($structure_tables as $table) {
                     $schemaonlies[] = "--table=$table";
                 }
-                $exec .= " && pg_dump --schema-only " . implode(' ', $schemaonlies) . $extra;
+                $exec .= " && $pg_dump_command --schema-only " . implode(' ', $schemaonlies) . $extra;
                 $exec .= (!$create_db && !$data_only ? ' --clean' : '');
             }
         }

```